src/config.h

```diff
@@ -74,6 +74,7 @@ struct Config {
     std::vector<NamedString> config_entries;  // Entries go into OrtSessionOptions::AddConfigEntry
 
     std::vector<ProviderOptions> provider_options;
+    std::vector<ProviderOptions> provider_options_backup;
```

Comment:
Do we have other config struct members that aren't part of the json? Currently the struct directly maps to the json.

Reply:
I added a new vector<string> that handles the providers.